### ISSUE TITLE
Add user contributions (was: Add a completely black background to the list of playable backgrounds.)

### ISSUE THREAD

**Issue by stavrossk** [stephanemagnenat/snakeme#1]
It would help the game visually (at least for me) the addition of fully black background (with white borders) to the list of playable backgrounds as it contrasts well with the snakes' white color.

**Comment by stephanemagnenat** [comment]
Do you mean a color theme, or an empty level?

**Comment by stavrossk** [comment]
@stephanemagnenat I mean a color theme, an smt file :)

**Comment by A1rPun** [comment]
In the past I've created this theme:

![BnW](https://i.imgur.com/GAJ0Zsz.png)

White background with black borders, if you are interested I can share it with you 😄

**Comment by stavrossk** [comment]
@A1rPun Yes please, share it with me, it looks very nice :)

**Comment by A1rPun** [comment]
The best I can do now is sharing through [files.fm](https://files.fm/u/xrs299kj).
I have an website from 2008 with many snakes and a couple more levels and themes (all with previews). I was busy putting it on my site but I have some hosting problems right now 😞.

**Comment by stavrossk** [comment]
@A1rPun What's the address of your website? Is it currently online?

**Comment by A1rPun** [comment]
The website I created 10 years ago is not online yet but it will be soon on [A1rPun.com/SnakeMe](http://a1rpun.com/snakeme/) :smile: .

**Comment by stephanemagnenat** [comment]
@A1rPun would you be interested if the game had a "contrib" feature, allowing to access contributions without any further installation just by checking a cross in a menu?

**Comment by A1rPun** [comment]
Hey @stephanemagnenat, what is the advantage of that feature over the "copy file to snakeme directory"? Is that feature downloading stuff without any copying needed?

Ps: I have also some contributions on the old ysagoon site

**Comment by stephanemagnenat** [comment]
@A1rPun indeed the idea would be to make them available to players without any additional downloads. Of course we could also imagine to make them available by default but then the list would be long and the individual contributions not well acknowledged. In the idea I suggest, I imagine a list of contributors with their nickname/name and a summary of their contribution, and a checkbox to make these available. Contributors could ask their contributions to be integrated into this list, with proper acknowledgements and maybe even a way to report feedback to them.

**Comment by A1rPun** [comment]
This sounds like a cool idea. Personally I would not implement "check checkbox to include authors contributions" but rather a "check checkbox to include this snake/level/theme". So you have a long list with contributions and you only check which ones you want to add to the game.

I think compiling a list with made contributions to this date is already a sufficient effort to play with cool new snakes, levels and themes IMO.

**Comment by stephanemagnenat** [comment]
I agree with your suggestion. Let's see if I find some times in the next months to implement that ;-)

**Comment by A1rPun** [comment]
I'm looking forward to that 😃. In the meantime.. here is a little teaser for some snakes I made which will be available when you made the change.

![snakes](https://i.imgur.com/7GPnxLj.png)

**Comment by stavrossk** [comment]
@stephanemagnenat So if I understand it correctly, all the contributions will be included in the SnakeMe package and only enabled if the user chooses them, or will the contributions be automatically downloaded by the program if the user chooses them?

**Comment by stephanemagnenat** [comment]
I would include them if authors agree, as it would make the code simpler and not require an Internet connection from the users. Anyway they are very small, and could be compressed if we want to further save space.

**Comment by stephanemagnenat** [comment]
One question, do we want a meta file per file giving a nice name and the author information, or do we want to somehow encode that in the file name to keep things simple?

**Comment by A1rPun** [comment]
Is it possible to add the metadata to the existing files without breaking backwards compatibility? If so I would like that.
Encoding inside a filename sounds spooky but can be a thing ofc  :smile:.

**Comment by stavrossk** [comment]
@stephanemagnenat While you are at it, could you please move .snk and .smt files to seperate folders for the SnakeMe folder to be a bit more tidy? Thanks :)

**Comment by stephanemagnenat** [comment]
@stavrossk agreed, I've opened an issue (#2).

**Comment by A1rPun** [comment]
My contributions are finally [on my site](http://a1rpun.com/snakeme/). Check them out 😛 
Some downloads won't work because of an case-sensitive issue. The files are there though.
